plugins/woocommerce/client/blocks/assets/js/blocks/product-filters/frontend.ts

```diff
@@ -224,4 +245,7 @@ const productFiltersStore = store( 'woocommerce/product-filters', {
 	},
 } );
 
-export type ProductFiltersStore = typeof productFiltersStore;
+export type ProductFiltersStore = {
```

Comment:
This type is not a good representation of the full store.

How about changing the destructuring to export the full type:

```ts
const productFiltersStore = store(...);

const { state, actions } = productFiltersStore;

export type ProductFiltersStore = typeof productFiltersStore;
```

Reply:
This is actually my first implementation, but Eslint complained about actions/state was used before it was defined. So I had to manual type the store to make it happy.

Reply:
I deleted that rule for the frontend files since it's not necessary.

- https://github.com/woocommerce/woocommerce/pull/56117/files#diff-ac2e79e01fc0519e24640ff7b3f71c2bc010d022770449cd91ccb93d6730d265R322-R327

Reply:
Great! I updated this PR to use type inference in d9424fc5061c